src/main/java/lotto/controller/LottoGameController.java

```diff
@@ -0,0 +1,88 @@
+package lotto.controller;
+
+import static lotto.configuration.IntegerConstants.COST_OF_LOTTO;
+
+import lotto.domain.entity.Lotto;
+import lotto.domain.entity.LottoGame;
+import lotto.domain.entity.LottoNumber;
+import lotto.domain.entity.Lottos;
+import lotto.domain.entity.ThousandUnitMoney;
+import lotto.domain.entity.WinningResult;
+import lotto.domain.dto.DtoMapper;
+import lotto.domain.dto.PurchaseDto;
+import lotto.domain.dto.WinningResultDto;
+import lotto.domain.entity.Purchase;
+import lotto.domain.entity.WinningTicket;
+import lotto.service.LottoGameService;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class LottoGameController {
+
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final LottoGameService lottoGameService;
+
+    public LottoGameController(InputView inputView,
+                               OutputView outputView,
+                               LottoGameService lottoGameService) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.lottoGameService = lottoGameService;
+    }
+
+    public void playGame() {
+        Purchase purchase = purchaseLotto();
+
+        outputView.outputPurchaseLottoList(DtoMapper.toPurchaseDtofrom(purchase.getLottosByDoubleList()));
+
+        WinningTicket winningTicket = generateWinningTicket();
+
+        WinningResult winningResult = lottoGameService.calcWinningResult(purchase, winningTicket);
+
+        outputView.outputWinningStatistics(
+                DtoMapper.toWinningResultDtofrom(
+                        winningResult.getRankings(), winningResult.getRateOfReturn()));
+
+        LottoGame lottoGame = LottoGame.create(purchase, winningTicket, winningResult);
+        Long lottoGameId = lottoGameService.saveLottoGame(lottoGame);
+
+        replayGame(false);
+    }
```

Comment:
개인적인 생각이지만 Controller에서 뷰의 역할을 맡는 것도 괜찮지만 엔티티를 사용하는 서비스에서 맡는 것도 괜찮을 것 같습니다. playGame()을 보시면 처음 읽을 때 잘 읽힌다는 느낌은 들지않았습니다. 아래와 같이 구현되도록 설계가 가능하다면 코드 가독성이 더욱 좋아질 것 입니다.

```suggestion
    public void playGame() {
        Purchase purchase = purchaseLotto();
        WinningTicket winningTicket = generateWinningTicket();
        WinningResult winningResult = lottoGameService.calcWinningResult(purchase, winningTicket);
        LottoGame lottoGame = LottoGame.create(purchase, winningTicket, winningResult);
        Long lottoGameId = lottoGameService.saveLottoGame(lottoGame);

        replayGame(false);
    }
```

Reply:
코드까지 직접 작성해서 예시를 보여주셔서 감사합니다 :)
덕분에 다음 미션에서는 더욱 가독성 좋은 컨트롤러를 만들 수 있겠네요 💪
확실히 아래 코드가 간결하기에 분리와 통합이 필요하다 생각합니다! 😆
하지만, 준호님께서 남겨주신 예시에는 아래와 같은 특징이 있기에 말씀해주신 **Service에서 UI를 담당하는** 대신 **Model과 입출력을 하나의 메소드로 묶는 것**이 더 좋은 해결 방법이라 생각이 되네요 :) 

- **Service가 View에 의존하는 것은 Layered Architecture 설계 원칙에 어긋납니다.**
Layered Architecture에 따른 MVC에서 Service는 Model에 속하며, 비니지스 계층에 해당됩니다.
View는 표현 계층으로, 비지니스 계층에서 표현 계층을 의존하는 것은 더 낮은 계층에서 더 높은 계층에 의존하는 것이 됩니다. 더 낮은 계층이 더 높은 계층에 의존하면, 출력 형태가 변경될 때마다 비즈니스 로직도 함께 변경해야 하는 문제가 생깁니다. 이는 유지보수를 어렵게 만들고, 코드의 유연성을 저하시킵니다.

저 또한 1주차에는 간결성을 위해 Service에서 View에 의존하도록 코드를 짰습니다!
하지만 피드백을 받고 Layered Architecture에 대해 공부해보니, Controller에서 View에 의존하는 것은 원칙을 어기고, View가 변경될 때마다 더 중요한 비지니스 로직을 건들여줘야 하는 문제가 있더라구요!
이와 관련해서 참고하시면 좋을 링크도 남겨놓겠습니다 :) 
https://korean-otter.tistory.com/226
https://www.oreilly.com/library/view/software-architecture-patterns/9781491971437/ch01.html

- **전체 흐름을 담당하는 controller에서 Model을 생성하고, 입출력을 담당하는 것은 자연스러운 현상**이지만, 말씀해주신 부분처럼 **아래 코드로 기능을 묶어주는 것이 더욱 가독성이 좋다**고 생각되네요 👍
아래와 같이 리팩터링 해봤습니다! 

~~~
public void playGame() {
    Purchase purchase = makePurchase();
    WinningTicket winningTicket = generateWinningTicket();
    WinningResult winningResult = calculateWinningResult(purchase, winningTicket);
    outputWinningStatistics(winningResult);
    saveLottoGame(purchase, winningTicket, winningResult);
    askToReplayGame();
}

private Purchase makePurchase() {
    Purchase purchase = purchaseLotto();
    outputView.outputPurchaseLottoList(DtoMapper.toPurchaseDtofrom(purchase.getLottosByDoubleList()));
    return purchase;
}

private WinningResult calculateWinningResult(Purchase purchase, WinningTicket winningTicket) {
    return lottoGameService.calcWinningResult(purchase, winningTicket);
}

private void outputWinningStatistics(WinningResult winningResult) {
    outputView.outputWinningStatistics(
            DtoMapper.toWinningResultDtofrom(
                    winningResult.getRankings(), winningResult.getRateOfReturn()));
}

private void saveLottoGame(Purchase purchase, WinningTicket winningTicket, WinningResult winningResult) {
    LottoGame lottoGame = LottoGame.create(purchase, winningTicket, winningResult);
    Long lottoGameId = lottoGameService.saveLottoGame(lottoGame);
}

private void askToReplayGame() {
    //...
}
~~~

덕분에 제 컨트롤러의 문제점을 알 수 있었습니다! 친절한 예시까지 정말 감사드려요! 저도 꼼꼼히 리뷰가도록 하겠습니다 👍👍👍

Reply:
피드백 해주신 내용 잘 읽어보았습니다. 변경 가능성의 관점에서 코드를 바라보면 서비스단에서 뷰에 의존하는 것이 안좋다는 것을 알게되었네요.

현재 하나의 컨트롤러에서 여러 private 메서드를 분리하셨는데 분리된 메서드를 각각의 컨트롤러로 다시 클래스로 만들어서 설계한다면 변경 가능성을 고려한 설계가 될 것 같습니다.